packages/backend/src/routes/checkout.js

```diff
@@ -1,80 +1,89 @@
 import express from 'express';
-import { db } from '../lib/database';
+import status from 'http-status';
+import { db, dbQueryPromise } from '../lib/database';
 import { getDistance } from '../lib/distanceMatrix';
+import { ticketTransaction } from '../lib/bank';
+import {
+  checkCreditCard,
+  validCreditCard,
+  getCardNumber,
+} from '../lib/creditcard';
+import { getSellerId, getPrice } from '../lib/tickets';
 
 const router = express.Router();
 
-router.post('/buy/submit', (req, res, next) => {
+router.post('/shipping/submit', async (req, res, next) => {
   if (req.session && req.session.userId) {
     const ticketInfo = {
-      boughtUserId: req.session.id, // check
       deliveryMethod: req.body.deliveryMethod,
       // address: req.body.address,
       ticketId: req.body.ticketId,
     };
 
-    db.query(
+    dbQueryPromise(
       'UPDATE tickets SET buyerId=?, deliveryMethod=?, available=0 WHERE id=?',
       [
-        ticketInfo.boughtUserId,
+        req.session.userId,
         ticketInfo.deliveryMethod,
         // ticketInfo.address,
         ticketInfo.ticketId,
-      ],
-      async (error, results, fields) => {
-        if (error) {
-          console.log(`Error contacting database: ${JSON.stringify(error)}`);
-          res.json(500, error);
-        } else {
-          // Get Distance
-          const distance = await getDistance(
-            ticketInfo.ticketId,
-            ticketInfo.boughtUserId
-          );
-          res.json(`The distance is: ${distance}`);
-        }
-      }
+      ]
+    ).catch(err =>
+      console.log(`Error contacting database: ${JSON.stringify(err)}`)
     );
+
+    // call function calculating shipping cost
+
+    // Get Distance
+    const distance = await getDistance(ticketInfo.ticketId, req.session.userId);
+    res.json(`The distance is: ${distance}`);
   } else res.json(401, 'Error: Not logged in');
+  // add to current price -database (display new price)
 });
 
-// not in database yet
-router.post('/payment/submit', (req, res, next) => {
+router.post('/billing/submit', async (req, res, next) => {
```

Comment:
Since this route executes the transaction, could we call it /submit (/checkout/submit) or /buy or /buy/submit? I feel like billing doesn't convey the execution of a transaction

Reply:
Done